src/queue.cpp

```diff
@@ -549,8 +549,8 @@ cl_int cvk_command_kernel::dispatch_uniform_region_iterate(
 
     size_t num_splitted_regions =
         ceil_div(num_workgroups[dim], vklimits.maxComputeWorkGroupCount[dim]);
-    size_t splitted_region_gws =
-        ceil_div(region.gws[dim], num_splitted_regions);
+    size_t splitted_region_gws = round_up(
```

Comment:
I haven't tried to verify this but it feels like gws calculations are getting more complex than they should be. Each region's gws will be a round multiple of the number of workgroups since we've already done non-uniform splitting so the gws calculation should be as simple as:
```
gws = N * lws
```

I suggest we refactor the logic to make this apparent as a separate PR.

Reply:
I think I got things too complicated because for some reason I wanted to have regions that are of the same size.
But it will definitely be easier to just make regions reach the limit, except for the last one.
```
size_t splitted_region_gws = vklimits.maxComputeWorkGroupCount[dim] * region.lws[dim];
```
We already have the `std::min` doing what we want to deal with the last region.

Reply:
Yup, easier to follow now, thanks!